jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -492,10 +501,18 @@ public boolean handle()
 
                     case COMPLETE:
                     {
-                        if (!_response.isCommitted() && !_request.isHandled() && !_response.getHttpOutput().isClosed())
+                        if (!_response.isCommitted())
                         {
-                            _response.sendError(HttpStatus.NOT_FOUND_404);
-                            break;
+                            if (!_request.isHandled() && !_response.getHttpOutput().isClosed())
+                            {
+                                // The request was not actually handled
+                                _response.sendError(HttpStatus.NOT_FOUND_404);
+                                break;
+                            }
+
+                            // Indicate Connection:close if we can't consume all.
+                            if (_response.getStatus() >= 200)
```

Comment:
I wonder whether this could be more lax... take into account 201 Created, 204 No Content, 206 Partial Content, etc.

Reply:
Why would 201, 204 and 206 not what this behaviour?   All we are doing is `consumeAll` a little earlier than we would have otherwise, so we get a chance to send a `Connection: close` if we can't consume all.     We just want to exclude partial responses (ie 100, 101 etc.) which handle content differently.

Reply:
Doh, wrong comment on my side.